### DISCUSSION TITLE
Sigmet and Volcanic Ash Advisories - filling volcano element.

### DISCUSSION THREAD

**Discussion by Mattonius5** [wmo-im/iwxxm#337] [Q&A]
Hello,
This affects All IWXXM versions, concerning VAA/Sigmet element "volcano/erupting volcano", especially "metce:eruptionDate", Given the VAA example below:

```html
<iwxxm:volcano>
    <metce:EruptingVolcano gml:id="uuid.1b9ee023-c87d-4fa0-bf9b-bb1c7215286c">
        <metce:name>KARYMSKY 1000-13</metce:name>
        <metce:position>
            <gml:Point gml:id="uuid.687c6e95-460a-4397-a602-ea8483acc3fc" srsDimension="2" axisLabels="Lat Long" srsName="http://www.opengis.net/def/crs/EPSG/0/4326">
                <gml:pos>54.03 159.27</gml:pos>
            </gml:Point>
        </metce:position>
        <metce:eruptionDate>2008-09-23T01:30:00Z</metce:eruptionDate>
    </metce:EruptingVolcano>
</iwxxm:volcano>
``` 
### Question 1, Volcanic Ash Advisory:
Element "iwxxm:volcano" requires "metce:EruptingVolcano" which then requires "name", "position" and "eruptionDate".
My understanding is, that "eruptionDate"  should be extracted from mandatory TAC element "Eruption details". Annex 3 allows for two options "Free text up to 64 characters" and "UNKNOWN".
I've collected extensive collection of Volcanic Ash Advisories in TAC form from all around the world. Around **60% of them do NOT contain** eruption date in any form inside the "Eruption details" element. In these instances, eruption date cannot be extracted, thus I cannot satisfy requirements of "metce:EruptingVolcano" by filling it's "eruptionDate".

Observations:

- When "eruptionDate" is missing, produced IWXXM **fails** schema validation.
- If I remove <iwxxm:volcano> element, Schematron validation fails on "VolcanicAshAdvisory.VolcanicAshAdvisory-3"
- Having empty <iwxxm:volcano/> passes validation.

What should I do in these three following cases?

1. generate translationFailed report
2. create empty element <iwxxm:volcano/> (this passes schema + schematron validation), however **information** about volcano name and position **are lost**.
3. Use date from field "DTG"
4. Use date from field "OBS/EST VA DTG"
5. Something else?

### Question 2 - Volcanic Sigmet
Volcanic Sigmet uses a similar element "iwxxm:eruptingVolcano" with very important distinction, it requires either element "metce:volcano" or it's substitution group "metce:eruptingVolcano". This way the name and position can be filled as part of complex type "VolcanoType" regardless presence of eruption date. If eruption date exists, "metce:eruptionVolcano" can be used (same as VAA). If eruption date does NOT exist, "metce:volcano" can be created. Thus we can pass all schema or schematron rules.

Why was not VAA modeled in the same fashion as Volcanic Sigmet in regards to "volcano" element?

### Questions 3 - Volcanic Sigmet
Considering the example for VAA on top, Sigmet looks almost identically, instead of <iwxxm:volcano>, there is <iwxxm:eruptingVolcano>, the rest is as described in question 2.
Annex 3 Table A6-1A, element Phenomenon for VA has following structure:

- [VA ERUPTION] (optional part)
- [MT volcano_name] (optional part)
- [PSN single_point_coordinate] (optional part)
- VA CLD (mandatory part)

What to do if Volcano name is not present? 

1. Create translationFailed
3. Use "UNNAMED" (VAA allowed name)
4. Use "UNKNOWN" (VAA allowed name)

Subsequent questions
What does the "VA ERUPTION" optional element means? My deductinon form online research is following:

- If present, it indicates that Volcano just erupted, thus I can convert it using "metce:eruptingVolcano" filling "eruptionDate" from "observation date". 
- If not present, eruption date is unknown, thus "eruptionDate" is absent, thus I shall use "metce:volcano" element.

APAC regional SIGMET guide states following:

1. If the erupting volcano is known and named then the following format is appropriate: **VA ERUPTION MT ASHVAL PSN <lat,lon> VA CLD**
2. If the location of the eruption is known but is un-named then the following format is appropriate: **VA ERUPTION PSN <lat,lon> VA CLD**
3. If the source of the volcanic ash is uncertain, then the following format is appropriate: **VA CLD**

In case 1, I would fully create element "metce:volcano" or "metce:eruptingVolcano" (depends on if any date should be added) like below:
```html
<iwxxm:eruptingVolcano>
    <metce:Volcano>
        <metce:name>MT HEKLA</metce:name>
        <metce:position>
            <gml:Point>
                <gml:pos>63.98 -19.67</gml:pos>
            </gml:Point>
        </metce:position>
    </metce:Volcano>
</iwxxm:eruptingVolcano>
``` 
In case 2, I would add to volcano:name "UNNAMED", like below:
```html
<iwxxm:eruptingVolcano>
    <metce:Volcano>
        <metce:name>UNNAMED</metce:name>
        <metce:position>
            <gml:Point>
                <gml:pos>63.98 -19.67</gml:pos>
            </gml:Point>
        </metce:position>
    </metce:Volcano>
</iwxxm:eruptingVolcano>
``` 
In case 3, I would NOT include <iwxxm:eruptingVolcano> in IWXXM as it's schema cardinality is 0-1 and no Schamatron rule enforces its presence.

Thanks for advise and guidance how should these elements be converted and please correct me where I am wrong.
Matous.

**Comment by blchoy** [comment]
Closed as now being handled under Issue #344
